File: frontend/src/lib/metadata.ts
import { api } from './api'

export interface RecordAgent { id: number; agentKind: string; agentId: number; agentName: string; role: string }
export interface RecordRelationship {
  id: number; sourceType: string; sourceId: number; targetType: string; targetId: number; type: string; targetTitle: string | null
}
export interface RecordActivity { workflowInstanceId: number; definitionName: string; status: string; startedAt: string }
export interface RecordMetadata {
  documentId: number
  documentNumber: string
  title: string
  agents: RecordAgent[]
  relationships: RecordRelationship[]
  activities: RecordActivity[]
}

export const metadataApi = {
  get: (docId: number) => api.get<RecordMetadata>(`/documents/${docId}/metadata`).then((r) => r.data),
}

export const AGENT_ROLE_LABELS: Record<string, string> = {
  Creator: 'منشئ', Owner: 'مالك', Custodian: 'الجهة الحافظة', Contributor: 'مساهم', Approver: 'معتمد', Recipient: 'مستلم',
}
export const AGENT_KIND_LABELS: Record<string, string> = { User: 'مستخدم', Position: 'منصب', OrgUnit: 'وحدة' }
export const REL_TYPE_LABELS: Record<string, string> = {
  IsVersionOf: 'إصدار من', References: 'يشير إلى', Supersedes: 'يحل محل', RespondsTo: 'رد على', PartOf: 'جزء من', Attachment: 'مرفق',
}

File: backend/src/Archiving.Domain/Entities/Metadata.cs
using Archiving.Domain.Common;

namespace Archiving.Domain.Entities;

/// <summary>The kind of agent acting on a record (ISO 23081 agent entity).</summary>
public enum AgentKind { User = 0, Position = 1, OrgUnit = 2 }

/// <summary>The role an agent plays in relation to a record.</summary>
public enum AgentRole { Creator = 0, Owner = 1, Custodian = 2, Contributor = 3, Approver = 4, Recipient = 5 }

/// <summary>A typed relationship between two records/entities (ISO 23081 relationship entity).</summary>
public enum RecordRelationshipType { IsVersionOf = 0, References = 1, Supersedes = 2, RespondsTo = 3, PartOf = 4, Attachment = 5 }

/// <summary>Links a record to an agent (person/seat/unit) with a role — ISO 23081 record↔agent metadata.</summary>
public class RecordAgent : BaseEntity
{
    public string RecordType { get; set; } = string.Empty;   // "Document" | "IncomingMail" | …
    public long RecordId { get; set; }
    public AgentKind AgentKind { get; set; }
    public long AgentId { get; set; }
    public AgentRole Role { get; set; }
}

/// <summary>A typed link between two records (or a record and a business activity) — ISO 23081 relationships.</summary>
public class RecordRelationship : BaseEntity
{
    public string SourceType { get; set; } = string.Empty;
    public long SourceId { get; set; }
    public string TargetType { get; set; } = string.Empty;
    public long TargetId { get; set; }
    public RecordRelationshipType Type { get; set; }
}
